hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AAnalyticsAcceleratorStream.java

```diff
@@ -0,0 +1,202 @@
+    IOStatistics ioStats;
+
+    try (FSDataInputStream inputStream = getFileSystem().open(dest)) {
+      ioStats = inputStream.getIOStatistics();
+      inputStream.seek(5);
+      inputStream.read(buffer, 0, 500);
+    }
+
+    verifyStatisticCounterValue(ioStats, STREAM_READ_ANALYTICS_OPENED, 1);
+  }
+
+ @Test
+ public void testMultiRowGroupParquet() throws IOException {
+    describe("A parquet file is read successfully");
+
+    Path dest = path("multi_row_group.parquet");
+
+   File file = new File("src/test/resources/multi_row_group.parquet");
+   Path sourcePath = new Path(file.toURI().getPath());
+   getFileSystem().copyFromLocalFile(false, true, sourcePath, dest);
+
+   FileStatus fileStatus = getFileSystem().getFileStatus(dest);
+
+   byte[] buffer = new byte[3000];
+   IOStatistics ioStats;
+
+   try (FSDataInputStream inputStream = getFileSystem().open(dest)) {
+     ioStats = inputStream.getIOStatistics();
+     inputStream.readFully(buffer, 0, (int) fileStatus.getLen());
+   }
+
+   verifyStatisticCounterValue(ioStats, STREAM_READ_ANALYTICS_OPENED, 1);
+ }
+
+  @Test
+  public void testConnectorFrameworkConfigurable() {
+    describe("Verify S3 connector framework reads configuration");
+
+    Configuration conf = getConfiguration();
+    removeBaseAndBucketOverrides(conf);
+
+    //Disable Predictive Prefetching
+    conf.set(ANALYTICS_ACCELERATOR_CONFIGURATION_PREFIX +
+        "." + LOGICAL_IO_PREFIX + ".prefetching.mode", "all");
+
+    //Set Blobstore Capacity
+    conf.setInt(ANALYTICS_ACCELERATOR_CONFIGURATION_PREFIX +
+        "." + PHYSICAL_IO_PREFIX + ".blobstore.capacity", 1);
+
+    ConnectorConfiguration connectorConfiguration =
+        new ConnectorConfiguration(conf, ANALYTICS_ACCELERATOR_CONFIGURATION_PREFIX);
+
+    S3SeekableInputStreamConfiguration configuration =
+        S3SeekableInputStreamConfiguration.fromConfiguration(connectorConfiguration);
+
+    Assertions.assertThat(configuration.getLogicalIOConfiguration().getPrefetchingMode())
+            .as("AnalyticsStream configuration is not set to expected value")
+            .isSameAs(PrefetchMode.ALL);
+
+    Assertions.assertThat(configuration.getPhysicalIOConfiguration().getBlobStoreCapacity())
+            .as("AnalyticsStream configuration is not set to expected value")
+            .isEqualTo(1);
+  }
+
+  @Test
+  public void testInvalidConfigurationThrows() throws Exception {
+    describe("Verify S3 connector framework throws with invalid configuration");
+
+    Configuration conf = getConfiguration();
+    removeBaseAndBucketOverrides(conf);
+    //Disable Sequential Prefetching
+    conf.setInt(ANALYTICS_ACCELERATOR_CONFIGURATION_PREFIX +
+        "." + PHYSICAL_IO_PREFIX + ".blobstore.capacity", -1);
+
+    ConnectorConfiguration connectorConfiguration =
+        new ConnectorConfiguration(conf, ANALYTICS_ACCELERATOR_CONFIGURATION_PREFIX);
+    Assertions.assertThatExceptionOfType(IllegalArgumentException.class)
```

Comment:
LambaTestUtils.intercept, as if an exception isn't raised you'll get the configuration.toString() of what was built